frame/balances/src/tests_reentrancy.rs

```diff
@@ -0,0 +1,339 @@
+						crate::Event::Transfer(
+							2,
+							3,
+							450,
+						),
+					),
+				),
+			);
+
+			assert!(
+				System::events().iter().any(
+					|er|
+					er.event == Event::pallet_balances(
+						crate::Event::DustLost(
+							2,
+							50,
+						),
+					),
+				),
+			);
+		}
+	);
+}
+
+#[test]
+fn transfer_dust_removal_tst2_should_work() {
+	ExtBuilder::default()
+		.existential_deposit(100)
+		.build()
+		.execute_with(|| {
+			// Verification of reentrancy in dust removal
+			assert_ok!(Balances::set_balance(RawOrigin::Root.into(), 1, 1000, 0));
+			assert_ok!(Balances::set_balance(RawOrigin::Root.into(), 2, 500, 0));
+
+			// In this transaction, account 2 free balance
+			// drops below existential balance
+			// and dust balance is removed from account 2
+			assert_ok!(Balances::transfer(RawOrigin::Signed(2).into(), 1, 450));
+
+			// As expected dust balance is removed.
+			assert_eq!(Balances::free_balance(&2), 0);
+
+			// Dust balance is deposited to account 1
+			// during the process of dust removal.
+			assert_eq!(Balances::free_balance(&1), 1500);
+
+			// Verify the events
+			// Number of events expected is 8
+			assert_eq!(System::events().len(), 9);
+
+			assert!(
+				System::events().iter().any(
+					|er|
+					er.event == Event::pallet_balances(
+						crate::Event::Transfer(
+							2,
+							1,
+							450,
+						),
+					),
+				),
+			);
+
+			assert!(
+				System::events().iter().any(
+					|er|
+					er.event == Event::pallet_balances(
+						crate::Event::DustLost(
+							2,
+							50,
+						),
+					),
+				),
+			);
+		}
+	);
+}
+
+#[test]
+fn repatriating_reserved_balance_dust_removal_should_work() {
+	ExtBuilder::default()
+		.existential_deposit(100)
+		.build()
+		.execute_with(|| {
+			// Verification of reentrancy in dust removal
+			assert_ok!(Balances::set_balance(RawOrigin::Root.into(), 1, 1000, 0));
+			assert_ok!(Balances::set_balance(RawOrigin::Root.into(), 2, 500, 0));
+
+			// Reserve a value on account 2,
+			// Such that free balance is lower than
+			// Exestintial deposit.
+			assert_ok!(Balances::reserve(&2, 450));
+
+			// Transfer of reserved fund from slashed account 2 to
+			// beneficiary account 1
+			assert_ok!(Balances::repatriate_reserved(&2, &1, 450, Status::Free), 0);
+
+			// Since free balance of account 2 is lower than
+			// existential deposit, dust amount is
+			// removed from the account 2
+			assert_eq!(Balances::reserved_balance(2), 0);
+			assert_eq!(Balances::free_balance(2), 0);
+
+			// account 1 is credited with reserved amount
+			// together with dust balance during dust
+			// removal.
+			assert_eq!(Balances::reserved_balance(1), 0);
+			assert_eq!(Balances::free_balance(1), 1500);
+
+			// Verify the events
+			// Number of events expected is 10
+			assert_eq!(System::events().len(), 10);
+
+			assert!(
+				System::events().iter().any(
+					|er|
+					er.event == Event::pallet_balances(
+						crate::Event::ReserveRepatriated(
```

Comment:
As I think I pointed out before, I think your (either personally or through your formatting config) are wrapping the lines at 60 or 80, which is too little. Configure this to be 100. 

Not an issue for this PR though.

Reply:
Thanks Corrected it :+1: